week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/product/Product.types.ts

```diff
@@ -0,0 +1,17 @@
+enum PRODUCT_TYPE {
+  buyNow = 'buyNow',
+  auction = 'auction',
+  free = 'free',
+}
+
+interface IProduct {
+  name: string;
+  amount: number;
+}
+
+interface IProductWithPrice extends IProduct {
+  price: number;
+}
```

Comment:
Te interface - są nadmiarowe. Masz już klasy które realizują to samo:
_są modelami danych_

Reply:
Trzymanie oddzielnie Interface i klas posiadających już modele danych będzie bardzo kosztowne, musisz zmieniać pola w kilku miejscach jeśli dodajesz lub usuwasz cokolwiek.

Pamiętaj że klasa to (type + value) a interface to (type) 
wiec same klasy wystarczą.